### PR TITLE
Add signUpWithEmailAndPassword

### PR DESCRIPTION
**Pull request by sarovin**

### CODE REVIEW

File: docs/_reference/auth.md [hunk 1]
```diff
@@ -204,6 +204,33 @@ methods:
...
+      - name: password
+        required: true
+        type: String
+        description: The user's password.
+    output: A [firebase.User](https://firebase.google.com/docs/reference/js/firebase.User.html) instance.
+    example: |
+      ```javascript
+      function* SignUpSaga(email, password) {
+        try {
+          const data = yield call(rsf.auth.signUpWithEmailAndPassword, email, password);
```
Comment (n6g7) [inline docs/_reference/auth.md]: Could you rename `data` to `user` please?

Comment (sarovin) [reply docs/_reference/auth.md]: of course

File: src/auth.js [hunk 2]
```diff
@@ -47,6 +47,11 @@ function * signInWithRedirect (authProvider) {
   yield call([auth, auth.signInWithRedirect], authProvider)
 }
 
+function * signUpWithEmailAndPassword (email, password) {
+  const auth = this.app.auth()
+  return yield call([auth, auth.signUpWithEmailAndPassword], email, password)
```
Comment (n6g7) [inline src/auth.js]: I can't find `signUpWithEmailAndPassword` in the [docs](https://firebase.google.com/docs/reference/js/firebase.auth.Auth), did you mean to use [`createUserWithEmailAndPassword`](https://firebase.google.com/docs/reference/js/firebase.auth.Auth#createUserWithEmailAndPassword) instead?

If so I think we should change the name of this saga as well to match the name of the firebase SDK method being called. :)

Comment (sarovin) [reply src/auth.js]: oops, ok i fix it ;)

**Comment by n6g7** [comment]
@sarovin thanks a lot, I published [v0.4.0](https://github.com/n6g7/redux-saga-firebase/releases/tag/v0.4.0) with your changes. :-)

**Pull request merged**